src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
+    private Object instance;
+    private Method method;
+
+    public DataProviderInformation(Method method, Object instance) {
```

Comment:
Having `IDataProviderAnnotation` included could help to find all information we need.